src/main/java/com/musala/atmosphere/commons/websocket/util/GsonUtil.java

```diff
@@ -0,0 +1,196 @@
+            try {
+                Type exceptionType = Class.forName(exceptionTypeName);
+
+                JsonObject jsonObject = g.fromJson(jsonMessage, JsonObject.class);
+                JsonElement exceptionData = jsonObject.get("exception");
+                Exception e = g.fromJson(exceptionData, exceptionType);
+                responseMessage.setException(e);
+            } catch (ClassNotFoundException e) {
+                // TODO: Error message, exception
+                e.printStackTrace();
+            }
+        }
+
+        return responseMessage;
+    }
+
+    private Type resolveTypeForRoutingAction(ResponseMessage responseMessage, RoutingAction routingAction) {
+        Type type = null;
+
+        switch (routingAction) {
+            case EXECUTE_SHELL_COMMAND_SEQUENCE:
+                type = new TypeToken<List<String>>() {
+                }.getType();
+                break;
+            case GET_SCREENSHOT:
+            case GET_DEVICE_LOGCAT:
+                type = new TypeToken<byte[]>() {
+                }.getType();
+                break;
+            case GET_UI_TREE:
+                type = new TypeToken<Tree<AccessibilityElement>>() {
+                }.getType();
+                break;
+            case GET_UI_ELEMENTS:
+            case GET_CHILDREN:
+            case EXECUTE_XPATH_QUERY_ON_LOCAL_ROOT:
+            case EXECUTE_XPATH_QUERY:
+                type = new TypeToken<List<AccessibilityElement>>() {
+                }.getType();
+                break;
+            case GET_LOGCAT_BUFFER:
+                type = new TypeToken<List<Pair<Integer, String>>>() {
+                }.getType();
+                break;
+            case GET_WEB_VIEWS:
+                type = new TypeToken<Set<String>>() {
+                }.getType();
+                break;
+            case FIND_WEB_ELEMENT:
+                type = new TypeToken<Map<String, Object>>() {
+                }.getType();
+                break;
+            case FIND_WEB_ELEMENTS:
+                type = new TypeToken<List<Map<String, Object>>>() {
+                }.getType();
+                break;
+            case WEB_ELEMENT_ACTION:
+                if (responseMessage.getTypeName() != null
+                        && responseMessage.getTypeName().equals(Pair.class.getName())) {
+                    type = new TypeToken<Pair<Integer, Integer>>() {
+                    }.getType();
+                    break;
+                }
+            default:
+                try {
+                    type = Class.forName(responseMessage.getTypeName());
+                } catch (ClassNotFoundException e) {
+                    // TODO: Error message, exception
+                    e.printStackTrace();
+                } catch (NullPointerException e1) {
+                    // nothing to do here
+                }
+                break;
+        }
+
+        return type;
```

Comment:
I find using the return statement once in a method less confusing than using it in several places.